content/blog/debugging-nodejs-programs-using-chrome/index.md

```diff
@@ -0,0 +1,308 @@
+
+![A showcase of a breakpoint on line 9 currently paused and a circle around the step into button](./step-inside.png)
+
+Let's say you're on line `9` and want to move into the `map` function. You can press the "step into" to move into line `10`.
+
+![The after effects of pressing the "step into" button after the screenshot above](./step-inside-part-2.png)
+
+Once inside the `map` function, there's even a button _to get you outside of that function and back to the parent caller's next line_. This might if you're inside of a lengthy `map` function, have debugged the line you wanted to inspect, and want to move past the `map` to the next line (the `console.log`). Doing so is as simple as "stepping in" a function, you simply press the "step outside" button to move to the next line
+
+![The "step outside" button being highlighted with a circle with the line 12 console log being paused](./step-outside.png)
+
+> While the example uses a callback in `map`, both of these "step into" and "step out of" also work on functions that are called. For example, assume the code was written as the following:
+>
+> ```javascript
+> const getEmployeeAges = partialList => {
+>   const ageArray = [];
+>   for (employee of partialList) {
+>     ageArray.push(employee.employee_age);
+>   }
+>   return ageArray;
+> };
+> 
+> app.get('/', (req, res) => {
+>   request('http://www.mocky.io/v2/5e1a9abe3100004e004f316b', (error, response, body) => {
+>     const responseList = JSON.parse(body);
+>     const partialList = responseList.slice(0, 20);
+>     const employeeAges = getEmployeeAges(partialList);
+>     console.log(employeeAges);
+>   });
+> });
+> ```
+>
+> You would still be able to "step into" `getEmployeeAges` and, once inside, "step outside" again in the same manor of the `map` function, as shown prior.
+
+# Saving Files {#editing-files-in-chrome}
+
+One more feature I'd like to touch on with the debugger before closing things out is the ability to edit the source files directly from the debugger. Using this feature, it can make the Chrome debugger a form of lite IDE, which may improve your workflow. So, let's revert our code to [the place it was at before we applied the fix we needed](#example-code) and go from there.
+
+![The screenshot of the debugger with the original code](./initial-debugger.png)
+
+Once this window is open, you're able to tab into or use your cursor to select within the text container that holds your code. Once inside, it should work just like a `textarea`, which _allows you to change code as you might expect from any other code editor_. Changing line `10` to `return employee.employee_age` instead of `return employee.employeeAge` will show an astererisk (`*`) to let you know your changes are not persisting. _Running your code in this state will not reflect the changes made to the code content on the screen_, which may have unintended effects.
```

Comment:
```suggestion
Once this window is open, you're able to tab into or use your cursor to select within the text container that holds your code. Once inside, it should work just like a `textarea`, which _allows you to change code as you might expect from any other code editor_. Changing line `10` to `return employee.employee_age` instead of `return employee.employeeAge` will show an asterisk (`*`) to let you know your changes have not yet been applied. _Running your code in this state will not reflect the changes made to the code content on the screen_, which may have unintended effects.
```